### PR TITLE
Integrate Cloudant with Spark Streaming

### PR DESCRIPTION
**Pull request by ykye**
1) Implement a custom receiver for Cloudant '_changes' API
2) Add Scala examples accordingly(Python isn't supported)
3) Update the documentation accordingly

BugzId: 61745

### CODE REVIEW

**Comment by ykye** [comment]
@mayya-sharipova

File: build.sbt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+lazy val cloudant_spark_sql = project.in(file("cloudant-spark-sql"))
+
+lazy val examples = project.in(file("examples/scala")).dependsOn(cloudant_spark_sql)
```
Comment (mayya-sharipova) [inline build.sbt:3]: What is the need for this file? Is it used only for eclipse?

It's also for packaging the example jar.
https://github.com/cloudant-labs/spark-cloudant/blob/61745-Spark-Streaming-integration/README_build.md#to-build
http://www.scala-sbt.org/0.13/docs/Multi-Project.html#Classpath+dependencies

**Comment by mayya-sharipova** [comment]
@ykye Kang Ye! Thanks very much for your work. The main code works without problem, and very well written. I have some small comments/questions just about examples.
1. During running `CloudantStreaming`, I have deleted a document in n_airportcodemapping, and got the following error: 16/03/09 10:02:10 ERROR JobScheduler: Error running job streaming job 1457535730000 ms.0
   org.apache.spark.sql.AnalysisException: Cannot resolve column name "airportName" among (_deleted, _id, _rev);
   at org.apache.spark.sql.DataFrame$$anonfun$resolve$1.apply(DataFrame.scala:152)
   
   Fixed
2. What is the need for files that you added: build.sbt, project/assembly.sbt, project/build.properties, project/eclipse.sbt? Can you please explain why we need these files. If they need only for Eclipse environment, I don't think that we need to add them here.  Since using Eclipse is optional, and people can use different environments.  
   
   We need to import `CloudantReceiver` in `CloudantStreaming`. One way is to put the `cloudant-spark.jar` into the folder `/examples/scala/lib` manually and the other way is to let the `examples` project depends on the project `cloudant-spark-sql`(http://www.scala-sbt.org/0.13/docs/Multi-Project.html#Classpath+dependencies). I select the second way to build the `examples`, so treat the `spark-cloudant` as the root project and the `examples`, `cloudant-spark-sql` as the sub-project of `spark-cloudant`, then we can set the dependence like this: `lazy val examples = project.in(file("examples/scala")).dependsOn(cloudant_spark_sql)`. We will build the `cloudant-spark.jar` in the same way(run `sbt assembly` in the `/cloudant-spark-sql` folder) but to build `spark_test_2.10-0.1-SNAPSHOT.jar`, we need to run `sbt package` in the `spark-cloudant` folder.
   We run `sbt eclipse` in the `spark-cloudant` folder as well. 
   
   https://github.com/cloudant-labs/spark-cloudant/blob/61745-Spark-Streaming-integration/README_build.md#to-build
3. We need to update `README_build.md`for building scala examples.  Basically, we need to say that before running `sbt package`, we need to create a folder `../examples/scala/lib`, and put `cloudant-spark.jar` into this folder. 
   
   We don't need to put `cloudant-spark.jar` into `/examples/scala/lib`
4. Do you want to add some information about Cloudant Streaming Receiver in`README.md` file the section `Implementation of Receiver` or  also in the section where you put your scala example? Just several lines - what is it, that it uses _changes feed and so on, so it would be helpful for users. 
   
   I have add some information on in the section `Implementation of Receiver` but add nothing in the `example section` because I saw there's no information in the other example sections. 

@mayya-sharipova

File: examples/scala/src/main/scala/mytest/spark/CloudantStreaming.scala [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    // Create the context with a 10 seconds batch size
+    val duration = new Duration(10000)
+    val ssc = new StreamingContext(sparkConf, duration)
+
+    val changes = ssc.receiverStream(new CloudantReceiver(Map(
+      "cloudant.host" -> "ACCOUNT.cloudant.com",
+      "cloudant.username" -> "USERNAME",
+      "cloudant.password" -> "PASSWORD",
+      "database" -> "n_airportcodemapping"), duration.milliseconds / 2))
+
```
Comment (mayya-sharipova) [inline examples/scala/src/main/scala/mytest/spark/CloudantStreaming.scala]: Why not putting this here:
```
val sc = ssc.sparkContext
val sqlContext = new SQLContext(sc)
```

What is the advantage of calling SQLContextSingleton?

**Comment by mayya-sharipova** [comment]
@ykye  Kang Ye, thanks for addressing my comments, and special thanks for your explanation. I understand now. And everything works fine as expected. +1 . You can merge the branch. Can you also please delete this branch after merging, and close the corresponding FB ticket.

**Pull request merged**